### PR TITLE
libimagentrylink: iter interface

### PR DESCRIPTION
**Pull request by matthiasbeyer**
First part of #802

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
@TheNeikos Also have a quick look here, please! :smile:

File: libimagentrylink/src/internal.rs [hunk 1]
```diff
@@ -89,8 +147,8 @@ impl InternalLinker for Entry {
         add_foreign_link(link, self.get_location().clone())
             .and_then(|_| {
                 self.get_internal_links()
-                    .and_then(|mut links| {
-                        links.push(new_link);
+                    .and_then(|links| {
+                        let links = links.chain(LinkIter::new(vec![new_link]));
```
Comment (TheNeikos) [inline libimagentrylink/src/internal.rs]: You can only put one type of objects into a vec, is this what you want here?

Comment (matthiasbeyer) [inline libimagentrylink/src/internal.rs]: Hm, not sure what you mean... `get_internal_links()` returns an iterator and `Iterator::chain()` adapts two iterators to become one... that's what it does here...

Comment (TheNeikos) [inline libimagentrylink/src/internal.rs]: Never mind, I misread ^^

**Comment by TheNeikos** [comment]
Looks alright to me

**Pull request merged**